### DISCUSSION TITLE
Working towards "Works with Home Assistant Partner" approval

### DISCUSSION THREAD

**Discussion by ikb42** [LightwaveSmartHome/homeassistant-lightwave-smart#25] [General]
We're excited to announce that we're actively working towards approval for the "Works with Home Assistant Partner" program. This is a significant step in the journey to better integrate with the Home Assistant ecosystem and provide a seamless experience for users.

We believe in the power of community and want to involve you all in this process.

- Feedback: We welcome your valuable feedback on the integration, any potential improvements, or any specific features you'd like to see.

- Contribute: If you're interested in getting involved, we encourage you to submit pull requests. Your contributions are highly appreciated and will help achieve this important milestone.

**Comment by xela1** [comment]
Hi @ikb42, I'll have a big think about this over the next few days, but I'll re-iterate the one key feature that's been requested (and at times promised) over the years and that is local access to the LinkPlus so we don't have to rely on Cloud. I don't know if this helps with approvals for Works with Home Assistant, but it will definitely go down better.

**Reply by adec** [reply]
Completely agree @xela1 - local access would be absolutely brilliant. I'm trying to switch as many integrations as possible to using local rather than cloud integrations. I know that it's possible to integrate with the switches using HomeKit but I'd really love to be able to do multi-press locally as well.

@ikb42: How can I help?  I'm not a developer by trade but have written code from time to time.  Is there a specification for the Link hub Local API that we can work with to try to introduce local access?

**Reply by ikb42** [reply]
hi @adec thanks for the offer, but at this stage there is no local API
It would be helpful though to test the latest [v3.5.36-Beta](https://github.com/LightwaveSmartHome/homeassistant-lightwave-smart/releases/tag/v3.5.36-Beta)

**Comment by AKWillows** [comment]
Is there any update on this? I'm looking at replacing the "old" version of the Lightwave plug in with this one, but the latest Betas seem to create a strange mix of two HA devices for each real device with the location of consumption/firmware/switch  varying depending on the Lightwave product type/ I'm loathed to fully switch across, but am really encouraged by the news that you are working towards the Works with HA" certification.

**Comment by Whizzykid** [comment]
@ikb42 have you made any progress on getting approval?

**Comment by AKWillows** [comment]
Agree, I was also wondering if there was any update. My home is entirely Lightwave sockets and switches. I see that LW are now advertise HA comparability on their website but with no full release for a while and various betas breaking with HA updates, it all feels quite flaky right now.

**Comment by ikb42** [comment]
Hi, we have made some limited progress.
[UPDATED] Testing the latest beta would really help move the process forward: [v3.5.37-Beta](https://github.com/LightwaveSmartHome/homeassistant-lightwave-smart/releases/tag/v3.5.37-Beta)

**Comment by xela1** [comment]
@ikb42 My understanding of the "Works with Home Assistant Partner" program is that local access is a requirement (cloud is optional), is this in Lightwave's roadmap now?

Have Lightwave done any analysis on your AWS monthly bandwidth costs due to Home Assistant usage, I'd predict that implementing a locally accessible websocket/api on the LinkPlus could probably save a large amount of monthly costs which could offset the costs for it's development.

**Comment by adec** [comment]
That’s an interesting point. Rather than providing a separate local API another approach would be to add button events to their HomeKit integration (or even better implement a Matter Bridge) which would have the advantage of benefiting more platforms. At present I’m using a combination of HomeKit and Lightwave cloud integration (via this integration) and my biggest bugbear is that I lose the ability to use button events when Internet connectivity is down.

**Comment by HenriPet** [comment]
https://support.lightwaverf.com/knowledge/how-to-set-up-lightwave-with-home-assistant

I had no idea lightwave are pointing people at this integration on their support page to integrate their product with Home Assistant! Surely there's opportunity there for it's future development?

**Reply by xela1** [reply]
> https://support.lightwaverf.com/knowledge/how-to-set-up-lightwave-with-home-assistant
> 
> I had no idea lightwave are pointing people at this integration on their support page to integrate their product with Home Assistant! Surely there's opportunity there for it's future development?

Yeah, it's been fairly recent (I suspect since they took on this fork of the code) but although @ikb42 has made some great progress with it, I feel that they're not putting in quite enough effort to be able to say that it 'works with home assistant'

**Reply by Patabugen** [reply]
I saw Lightwave take an active interest in Home Assistant and felt some optimism! Their techies are pretty good in my experience, so hopefully the company invests some time

**Comment by nicky2702** [comment]
another upvote for local-control without internet-access being required.  Internet in our village has semi-regular outages & I get funny looks from the mrs & kids when things wont work - turns out those multi-press events are highly-used!

**Comment by adec** [comment]
This gets my vote too as I’m about to decide whether to rip out the Lightwave switches and replace them with Shelly devices. I don’t want to invest in further Lightwave switches without local control including multi-press.
